micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jpa/HibernateQueryMetrics.java

```diff
@@ -0,0 +1,175 @@
+import io.micrometer.core.lang.NonNullFields;
+import org.hibernate.SessionFactory;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.event.service.spi.EventListenerRegistry;
+import org.hibernate.event.spi.EventType;
+import org.hibernate.event.spi.PostLoadEvent;
+import org.hibernate.event.spi.PostLoadEventListener;
+import org.hibernate.stat.QueryStatistics;
+import org.hibernate.stat.Statistics;
+
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A {@link MeterBinder} implementation that provides Hibernate query metrics. It exposes the
+ * same statistics as would be exposed when calling {@link Statistics#getQueryStatistics(String)}.
+ *
+ * @author Pawel Stepien
+ */
+@NonNullApi
+@NonNullFields
+public class HibernateQueryMetrics implements MeterBinder {
+
+    private static final String SESSION_FACTORY_TAG_NAME = "entityManagerFactory";
+
+    private final Iterable<Tag> tags;
+
+    private final SessionFactory sessionFactory;
+
+    /**
+     * Create {@code HibernateQueryMetrics} and bind to the specified meter registry.
+     *
+     * @param registry           meter registry to use
+     * @param sessionFactory     session factory to use
+     * @param sessionFactoryName session factory name as a tag value
+     * @param tags               additional tags
+     */
+    public static void monitor(MeterRegistry registry, SessionFactory sessionFactory, String sessionFactoryName, String... tags) {
+        monitor(registry, sessionFactory, sessionFactoryName, Tags.of(tags));
+    }
+
+    /**
+     * Create {@code HibernateQueryMetrics} and bind to the specified meter registry.
+     *
+     * @param registry           meter registry to use
+     * @param sessionFactory     session factory to use
+     * @param sessionFactoryName session factory name as a tag value
+     * @param tags               additional tags
+     */
+    public static void monitor(MeterRegistry registry, SessionFactory sessionFactory, String sessionFactoryName, Iterable<Tag> tags) {
+        new HibernateQueryMetrics(sessionFactory, sessionFactoryName, tags).bindTo(registry);
+    }
+
+    /**
+     * Create a {@code HibernateQueryMetrics}.
+     *
+     * @param sessionFactory     session factory to use
+     * @param sessionFactoryName session factory name as a tag value
+     * @param tags               additional tags
+     */
+    public HibernateQueryMetrics(SessionFactory sessionFactory, String sessionFactoryName, Iterable<Tag> tags) {
+        this.tags = Tags.concat(tags, SESSION_FACTORY_TAG_NAME, sessionFactoryName);
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+
+        registerListeners(registry);
+
+    }
+
+    /**
+     * Register hibernate loadEvent listener - event trigger the metrics registration,
+     * only select queries are recorded in QueryStatistics {@link org.hibernate.stat.spi.StatisticsImplementor}
```

Comment:
If we think it would be helpful to users to have this information, we should add it to one of the public methods or the class JavaDoc. On a private method, it will not be noticed by most users.